tests/test_core.py

```diff
@@ -23,3 +35,17 @@ def test_enums():
         assert "Camera" in mmc.getLoadedDevices()
         cfg = mmc.getConfigState("Channel", "DAPI")
         assert isinstance(cfg, pmn.Configuration)
+
+
+@pytest.mark.skipif(sys.platform != "win32", reason="DLL works only on Windows")
+def test_camera_device(demo_camera_dll: str) -> None:
+    """Tests that the built core can load a camera device from a local DLL."""
+    logger.info("Testing camera device loading; path: %s", demo_camera_dll)
+    mmc = pmn.CMMCore()
+    mmc.setDeviceAdapterSearchPaths([demo_camera_dll])
```

Comment:
this needs to point to the directory that contains the dll, not the dll itself, I think that's why the test is failing?

Let's make something that looks like this:

```
tests
├── adapters
│   ├── darwin/
│   ├── linux/
│   └── win32/
└── test_core.py
```

and then make a fixture:

```python
@pytest.fixture
def adapter_paths() -> list[str]:
    adapters = Path(__file__).parent / "adapters" / sys.platform
    if len(list(adapters.glob("*"))):
        return [str(adapters)]
    return []

def test_something(adapter_paths):
    mmc = pmn.CMMCore()
    mmc.setDeviceAdapterSearchPaths(adapter_paths)
```

Reply:
or actually... since I think git will give you a hard time about empty folders, just do:

```
adapters = Path(__file__).parent / "adapters" / sys.platform
if adapters.exists(): ...
```